src/Adapter/Category/CommandHandler/EditRootCategoryHandler.php

```
@@ -44,6 +45,7 @@ final class EditRootCategoryHandler extends AbstractObjectModelHandler implement
      *
      * @throws CannotEditCategoryException
      * @throws CategoryNotFoundException
+     * @throws CannotEditRootCategoryException
      */
     public function handle(EditRootCategoryCommand $command)
```

So the whole handler is now forbidden to be used Maybe then we could throw an exception in the Command constructor itself since it's not supposed to be used any more?

---

That will make me get the Catgory Object in the command. It's not ... usual

---

👍  indeed I dont see how we can check "is category a root category" without fetching the data from database, and this cannot be done inside a Command 😢